apps/src/code-studio/components/header/HeaderFinish.jsx

```diff
@@ -0,0 +1,94 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import $ from 'jquery';
+
+const styles = {
+  headerContainer: {
+    position: 'relative',
+    overflow: 'hidden',
+    height: 40
+  },
+  headerInner: {
+    position: 'absolute'
+  },
+  scriptLinkWithUpdatedAt: {
+    display: 'block'
+  },
+  outerContainer: {
+    textAlign: 'right'
+  },
+  containerWithUpdatedAt: {
+    verticalAlign: 'bottom',
+    whiteSpace: 'nowrap',
+    overflow: 'hidden',
+    textOverflow: 'ellipsis',
+    display: 'inline-block'
+  },
+  headerVignetteRight: {
+    position: 'absolute',
+    width: '100%',
+    height: 18,
+    pointerEvents: 'none',
+    background:
+      'linear-gradient(to right, rgba(0, 173, 188, 0) calc(100% - 20px), rgba(0, 173, 188, 1) 100%)'
+  }
+};
+
+export default class HeaderFinish extends React.Component {
+  static propTypes = {
+    lessonData: PropTypes.object,
+    width: PropTypes.number,
+    setDesiredWidth: PropTypes.func
+  };
+
+  setDesiredWidth() {
```

Comment:
I tend to find name overloads like this pretty confusing and can lead to odd bugs if someone is trying to rename the prop. Perhaps the prop should be called `reportDesiredWidth`?

Reply:
I thought about this a bunch but couldn't come up with something better.  `report` sounds like it refers to a reporting system?  At the end of the day, it is a `set` function, even if the call does come from one of its children...